api/controllers/game/resolve.js

```diff
@@ -172,6 +172,7 @@ module.exports = function (req, res) {
                     newSecondCard = _.sample(game.deck).id;
                   }
                   gameUpdates.secondCard = newSecondCard;
+                  cardsToRemoveFromDeck.push(newSecondCard);
```

Comment:
1. You're definitely right about breaking this up. I think we likely want one file/function per card effect and for this one to just delegate to the appropriate helper.
2. Good catch on the while loop...this one is actually unnecessary! This was copied from the logic for drawing two cards, but it's simpler here because you only draw one. If you look on lines 137-150, you can get the idea of what this was emulating, but again it's unnecessary here because we're only drawing one card. I'll remove that loop altogether b/c it's garbage here.

But for the record the idea here was that when the 5 draws _two cards_ (again not the case for this flow), the topCard and secondCard are put into the player's hand. Then if there's at least one card left in the deck, we sample from the deck to pick a new topCard. Then we if there's still another card in the deck, we sample again to pick the new secondCard. For the secondCard we want to make sure that we don't accidentally pick the same card we chose for the newTopCard, so the while loop is intended to make sure the secondCard selection is a new card that hasn't been drawn.

That said the while loop is a dumb way of doing it. What would be better would be to sample by index rather than getting the object reference for the card in the deck _then remove the card from the deck in memory_ before sampling the second time, so that you can safely assume the second selection won't collide. Then there's no need for a while loop.

Think I should fix that while I'm in here or leave this update smaller? Imo it makes sense to rework this part of the logic for clarity's sake since it still won't be a huge delta

Reply:
I refactored to remove the while loops from 5 resolution

Reply:
Awesome! Thanks for the explanation. I would fork the net new work to a new ticket though just to keep a separation of concerns.

All of that makes sense but I have a question about how you maintain randomness of the deck for a new game. Does sample work to pick a random card or are you actually shuffling the deck per game and tracking a net new dictionary of the card order in which case wouldn’t sample not be required?

Reply:
So the deck is basically unordered. There are two cards topCard and secondCard that are explicitly specified and then the other cards are just "in the deck" until we need one to become the topCard or secondCard at which point we sample one randomly. 

I refactored to remove the while loops bc it's just a few lines and it's relevant to the current task even if it's separable. Does the new implementation make more sense?

Reply:
I'll add that when I say the deck is unordered, that what I mean is that while the actual game.deck attribute in memory is an array, that the order of cards in game.deck is not guaranteed to be consistent across different times we query the database to construct the game. So it technically it has an order, it's just that we can't depend on the order for logic because it's not persistent